functions/go/ensure-name-substring/README.md

```diff
@@ -0,0 +1,100 @@
+# ensure-name-substring
+
+### Overview
+
+<!--mdtogo:Short-->
+
+Ensure a name substring is part of the name.
+
+<!--mdtogo-->
+
+### Synopsis
+
+<!--mdtogo:Long-->
+
+Note: This is an alpha function, and we are actively seeking feedback on the
+function config syntax and behavior. If you have suggestion or feedback, please
+file an issue [here](https://github.com/GoogleContainerTools/kpt/issues/new/choose).
+
+If the desired name substring is already part of the name, it takes no actions.
+Otherwise, it prepends or appends the name substring.
+
+Note: If the original name of a resource happens to contain the desired
+substring, the desired substring will not be added again. Users need to ensure
+the name collisions don't happen.
+
+To configure it using a ConfigMap, only one key-value pair is allowed in `data`
+field. The key must be one of `prepend` and `append`, and the value is the
+desired name substring. 
+
+```yaml
+prepend|append: Desired name substring
```

Comment:
Can we have an option to restrict the position of substring? For example I only want to ensure substring `dev-` in the beginning of the name.

Reply:
I guess we can do that in another function "ensure-name-prefix" which is approach 1 in go/kpt-fn-set-name-prefix-simplified

Reply:
ack